src/shell/Shell.cpp

```diff
@@ -290,7 +290,31 @@ static Trap::TrapResult executeWASM(Store* store, const std::string& filename, c
     Walrus::Trap trap;
     return trap.run([](ExecutionState& state, void* d) {
         RunData* data = reinterpret_cast<RunData*>(d);
-        data->module->instantiate(state, data->importValues);
+        Instance* instance = data->module->instantiate(state, data->importValues);
+
+        for (auto&& exp : data->module->exports()) {
+            if (exp->exportType() == ExportType::Function) {
+                if ("_start" != exp->name()) {
```

Comment:
I've added a simple check if any wasi functions were imported, it will only run this if there were any. 

In terms of testing, it may affect performance a bit, but I dont see any significant performance loss with this.

Reply:
I am not worried about performance, I worried that the `_start` might be provided by some test. But no core regression test imports wasi, so this is better.